sqlalchemy-stubs/ext/asyncio/result.pyi

```diff
@@ -1,73 +1,91 @@
 from typing import Any
-from typing import Int
 from typing import Iterator
 from typing import List
 from typing import Mapping
 from typing import Optional
+from typing import TypeVar
 
-from ... import util as util
-from ...engine.result import FilterResult as FilterResult
-from ...engine.result import FrozenResult as FrozenResult
-from ...engine.result import MergedResult as MergedResult
-from ...engine.result import Row as Row
-from ...util.concurrency import greenlet_spawn as greenlet_spawn
+from ...engine.result import FilterResult
+from ...engine.result import FrozenResult
+from ...engine.result import MergedResult
+from ...engine.result import Result
+from ...engine.result import RMKeyView
+from ...engine.result import Row
+
+_TAsyncResult = TypeVar("_TAsyncResult", bound=AsyncResult)
+_TAsyncScalarResult = TypeVar("_TAsyncScalarResult", bound=AsyncScalarResult)
+_TAsyncMappingResult = TypeVar(
+    "_TAsyncMappingResult", bound=AsyncMappingResult
+)
 
 class AsyncCommon(FilterResult):
     async def close(self) -> None: ...
 
 class AsyncResult(AsyncCommon):
-    def __init__(self, real_result: Any) -> None: ...
-    def keys(self): ...
-    def unique(self, strategy: Optional[Any] = ...): ...
-    def columns(self, *col_expressions: object) -> AsyncResult: ...
+    def __init__(self, real_result: Result) -> None: ...
+    def keys(self) -> RMKeyView: ...
+    def unique(
+        self: _TAsyncResult, strategy: Optional[Any] = ...
+    ) -> _TAsyncResult: ...
+    def columns(
+        self: _TAsyncResult, *col_expressions: object
```

Comment:
what's the difference between Any and object?

Reply:
In the case of arguments, not much since both `Any` and `object` accept all types. In the case of return types and variable annotations, `Any` tells the type checker to ignore the type and `object` would tell the type checker that the only operations that are permitted on the value are those that `object` defines.

Reply:
understood. for consistency I think it should be Any, since I don't think object is used anywhere else I've looked. But I'm fine with it either way